### DISCUSSION TITLE
ImageMagick produce sometime corrupted image

### DISCUSSION THREAD

**Discussion by Zeus64** [ImageMagick/ImageMagick#2670] [Development]
Hello,

Today I faced a very strange problem, image magick produced a totally corrupted image. This is very rare, and I can not even reproduce it, however I can see the result

To resume :
* imagemagick 7.0.7.23
* I do all operations in background thread
* I do many CloneMagickWand
* I do MagickResizeImage
* all operation return MagickTrue
* I do MagickGetImageBlob to retrieve the bytes
* I do DestroyMagickWand
* I save the bytes on the disk and sometime (very rarely and not reproducible) saved image are totally corrupted (random byte or just byte 0 everywhere in the file)
* corrupted image with random bytes have exactly the same size than if the image was not corrupted
* at the end of my process I do MagickRelinquishMemory to release the bytes

It's look like the bytes I get from MagickGetImageBlob get corrupted before I physically save it to the hard drive and also before I call MagickRelinquishMemory.  

Is this a know bug ?

**Comment by snibgo** [comment]
I suspect a bug in your program. If you paste it here, or link to the source code, perhaps someone might spot the bug.

**Reply by Zeus64** [reply]
Thanks @snibgo ! the code is very complicated, and written in delphi. I just first want to be sure that it's was not a know bug in ImageMagick. As this happen very (very) rarely (I see it only one time in 2 years) it's very hard (impossible even) to reproduce it :(

**Comment by fmw42** [comment]
Perhaps you have run out of tmp space and ImageMagick then crashed and left the image corrupt.
